client/blocks/reader-join-conversation/dialog.jsx

```diff
@@ -0,0 +1,44 @@
+import { Dialog } from '@automattic/components';
+import { Button } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import useLoginWindow from 'calypso/components/login-window';
+import './style.scss';
+
+const ReaderJoinConversationDialog = ( { onClose, isVisible } ) => {
+	const translate = useTranslate();
+
+	const { login, createAccount } = useLoginWindow( {
+		onLoginSuccess: () => window.location.reload(),
+		domain: 'wpcalypso.wordpress.com', //Need to use this while testing locally (also needs to be sandboxed)
```

Comment:
Would this come from environment config in the future?

Reply:
Not sure. Ideally we can set the domain to work locally and work in production but I haven't figured out how to do this yet.

Reply:
Added a config check here as suggested